### PR TITLE
ENG-39 fix directive data sharing among multiple instances

### PR DESCRIPTION
**Pull request by thilinasan16**
Fix the issue on directive instances overriding same values on comparison.
## Checklist

- [ ] Test Coverage is 100% for the newly added code
- [ ] Storybook stories are added/updated for the changed areas
- [x] Components standards defined [in this document](https://docs.google.com/document/d/16_Nd3VxE_lTD9pVkONQ0egn7IiwyX1pVXZjzl-V4tU8/) are followed
- [x] Code is linted properly
- [x] Developer testing is done for the affected areas
- [ ] Package version updated (not applicable to ent branch)
- [ ] Changelog.md updated on possible breaking (applicable to ent branch)

### CODE REVIEW

File: components/src/directives/focus-first-element/index.ts [hunk 1]
```diff
@@ -9,28 +9,43 @@ const excludeElements =
...
 
 const focusonFirstElementDirective: Directive = {
-  updated(el: FocusFirstHTMLElement) {
+  updated(el: FocusFirstHTMLElement, binding, vnode: VNode) {
     updatedFocusFirstElement = el.querySelectorAll(
       focusableElements + ', ' + excludeElements,
     )[0];
-    if (firstFocusableElement !== updatedFocusFirstElement) {
-      focusOnFirstElement(el, focusableElements + ', ' + excludeElements);
+    firstFocusedElementsOnUpdated.set(vnode.scopeId, updatedFocusFirstElement);
```
Comment (Super-Chama) [inline components/src/directives/focus-first-element/index.ts:32]:
```
 updated(el: FocusFirstHTMLElement, binding, vnode: VNode) {
  const elm = el.querySelectorAll(focusableElements + ', ' + excludeElements)[0];
  firstFocusedElementsOnUpdated.set(vnode.scopeId, elm);
  if (
    firstFocusedElementsOnMounted.has(vnode.scopeId) &&
    elm !== firstFocusedElementsOnUpdated.get(vnode.scopeId)
  ) {
    focusOnFirstElement(
      el,
      focusableElements + ', ' + excludeElements,
      vnode,
    );
  }
};
```

File: components/src/directives/focus-first-element/index.ts [hunk 2]
```diff
@@ -9,28 +9,43 @@ const excludeElements =
   'button:not(.oxd-dialog-close-button,.modal-reset-button)';
 let firstFocusableElement: Element;
```
[the remark this answers is not in the document]
Comment (thilinasan16) [reply components/src/directives/focus-first-element/index.ts]: removed global variables

File: components/src/directives/focus-first-element/index.ts [hunk 3]
```diff
@@ -9,28 +9,43 @@ const excludeElements =
...
 let updatedFocusFirstElement: Element;
+const firstFocusedElementsOnMounted = new Map();
+const firstFocusedElementsOnUpdated = new Map();
```
[the remark this answers is not in the document]
Comment (thilinasan16) [reply components/src/directives/focus-first-element/index.ts]: updated.

File: components/src/directives/focus-first-element/index.ts (hunk 1 above)
Comment (thilinasan16) [reply components/src/directives/focus-first-element/index.ts:30]: guess here the suggested comparison is invalid as need to compare between mounted and updated element of same scopeId.

**Pull request merged**